mimic/model/flavor_objects.py

```diff
@@ -1,279 +1,107 @@
+            if flavor.flavor_id == flavor_id:
+                return flavor
 
-    def extra_specs_json(self):
+    def create_flavors_list(self):
         """
-        Create a JSON-serializable data structure describing
-        ``OS-FLV-WITH-EXT-SPECS:extra_specs`` for a flavor.
+        Generates the data for each flavor in each flavor class
         """
-        return {
-            "class": "memory1",
-            "policy_class": "memory_flavor"
-        }
+        rackspace_flavors = [RackspaceStandardFlavor, RackspaceComputeFlavor,
+                             RackspacePerformance1Flavor, RackspaceOnMetalFlavor,
+                             RackspacePerformance2Flavor, RackspaceMemoryFlavor,
+                             RackspaceIOFlavor, RackspaceGeneralFlavor]
 
+        if len(self.flavors_store) < 1:
+            for flavor_class in rackspace_flavors:
+                for flavor, flavor_spec in flavor_class.flavors.iteritems():
+                    flavor_name = flavor
+                    flavor_id = flavor_spec['id']
+                    ram = flavor_spec['ram']
+                    vcpus = flavor_spec['vcpus']
+                    network = flavor_spec['rxtx_factor']
+                    disk = flavor_spec['disk']
+                    tenant_id = self.tenant_id
+                    flavor = flavor_class(flavor_id=flavor_id, tenant_id=tenant_id,
+                                          name=flavor_name, ram=ram, vcpus=vcpus,
+                                          rxtx=network, disk=disk)
+                    self.flavors_store.append(flavor)
 
-class RackspaceOnMetalFlavor(Flavor):
-    """
-    A Rackspace onMetal flavor object representation
-    """
-    flavors = {"OnMetal Compute v1": {"id": "onmetal-compute1", "ram": 32768, "vcpus": 20,
-                                      "rxtx_factor": 10000, "disk": 32},
-               "OnMetal IO v1": {"id": "onmetal-io1", "ram": 131072, "vcpus": 40,
-                                 "rxtx_factor": 10000, "disk": 32},
-               "OnMetal Memory v1": {"id": "onmetal-memory1", "ram": 524288, "vcpus": 24,
-                                     "rxtx_factor": 10000, "disk": 32}}
-
-    def extra_specs_json(self):
+    def list_flavors(self, include_details, absolutize_url):
         """
-        Create a JSON-serializable data structure describing
-        ``OS-FLV-WITH-EXT-SPECS:extra_specs`` for an onMetal flavor.
+        Return a list of flavors with details.
         """
-        return {
-            "quota_resources": "instances=onmetal-compute-v1-instances,ram=onmetal-compute-v1-ram",
-            "class": "onmetal",
-            "policy_class": "onmetal_flavor"
-        }
+        if len(self.flavors_store) < 1:
+            self.create_flavors_list()
+        flavors = []
+        for flavor in self.flavors_store:
+            if self.region_name != "IAD" and isinstance(flavor, RackspaceOnMetalFlavor):
+                continue
```

Comment:
non-blocking, since I see this was here before: This seems like a piece of business logic awkwardly wrapped up in a container with very little business logic around it. I wonder if it could be handled more elegantly somewhere else.